sdk/core/azure-core/src/test/java/com/azure/core/implementation/AsynchronousFileChannelAdapterTest.java

```diff
@@ -100,11 +98,11 @@ public void testReadWithCallbackAndOffset() throws IOException, InterruptedExcep
             latch.await(60, TimeUnit.SECONDS);
         }
         readData.flip();
-        assertArrayEquals(Arrays.copyOfRange(data, offset, data.length), readData.array());
+        assertArraysEqual(data, offset, data.length - offset, readData.array(), data.length - offset);
     }
 
     private static void readWithCallback(AsynchronousByteChannel channel, ByteBuffer aggregator, CountDownLatch latch) {
-        ByteBuffer buffer = ByteBuffer.allocate(1 + RANDOM.nextInt(127));
+        ByteBuffer buffer = ByteBuffer.allocate(1 + ThreadLocalRandom.current().nextInt(127));
```

Comment:
nit: would be nice to have utils that return random in core tests and checkstyle warn on random usage in tests